### ISSUE TITLE
[struct2depth]Resource exhausted with batch size=1

### ISSUE THREAD

**Issue by anakita** [tensorflow/models#8455]
Great work @aneliaangelova and @VincentCa  and team. I am getting the following error whenever I use masks also for training along with the rgb images:

Resource exhausted: OOM when allocating tensor with shape[1,4,53248]
I have a 10gb gpu .The model trains perfectly without masks ,i.e without handle motion.
I got this error for a batchsize of 1.
Can anyone please help me out.
This is the mask stack which I give as input:
![0000000001-fseg](https://user-images.githubusercontent.com/64612239/80753490-f99edc00-8b4a-11ea-9d03-81482437f9a9.jpg)
I also tried eliminating small masks but end up with the same error.
Thank you

**Comment by anakita** [comment]
@aneliaangelova and @VincentCa can you please provide me an update on the above issue.The error occurs only in case of some images even though they are aligned by using alignment.py.There are around 2000 images,and I manually inspected the images,and I conclude that it is not due to memory exhausting but it occurs in case if the images are not aligned perfectly.But it is difficult to make this out even visualiziing each image manually.
Thankyou

**Comment by VincentCa** [comment]
Hi @anakita 
Please double-check the IDs in your masks, you can do this programmatically. In particular, if an object ID is part of one image, it needs to be in the entire frame triplet. That is, if you have an object not associated to the same ID across all frames in the sequence, it should be dropped in all of them.

Perfect alignment is not necessary. But also, please make sure to only use PNG masks as JPEG would be lossy and introduce compression artifacts changing object IDs.

Hope this helps!
Vincent

**Comment by anakita** [comment]
Thanks a lot for the reply.I will check out the saving format of the images as well.
Thank you

**Comment by sgasperini** [comment]
hi @anakita !
I'm experiencing the same issue.
Did you manage to make it train with the masks (and motion model) eventually?
thanks in advance!
